File: NEETCODE/Easy/Contains_Duplicate.py
# https://neetcode.io/problems/duplicate-integer

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        dic = set()
        ans = False
        for n in nums:
            if n not in dic:
                dic.add(n)
            else:
                ans = True
                break
        return ans

File: NEETCODE/Easy/Contains_Duplicate.js
// https://neetcode.io/problems/duplicate-integer

class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const numSet = new Set(nums)
        return nums.length !== numSet.size
    }
}